### PR TITLE
Let users use v1 without additional steps

### PR DESCRIPTION
**Pull request by Szymon20000**
Fixes: https://github.com/software-mansion/react-native-reanimated/issues/1838
Fixes: https://github.com/software-mansion/react-native-reanimated/issues/1825
Fixes: https://github.com/software-mansion/react-native-reanimated/issues/1798
We forgot To wrap some initialization steps of rea 2 into a condition that checks if the native side of reanimated 2 is available. Because of that people that want to use only v1 have to go through installation steps on Android. 

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Added TS types tests
- [ ] Added unit / integration tests
- [ ] Updated documentation
- [ ] Ensured that CI passes

### CODE REVIEW

**Review by piaskowyk — APPROVED** [review]
I tested and is OK. 
Maybe we should change little the warn message https://github.com/software-mansion/react-native-reanimated/blob/4ff075eb8c59581642de1aded30db58af7163b8f/src/reanimated2/core.js#L25 to
`Reanimated 2 failed to create a worklet, maybe you forgot to add Reanimated's babel plugin? If you want to use only Reanimated v1, you can ignore this message.`

File: src/reanimated2/core.js [hunk 1]
```diff
@@ -15,14 +15,13 @@ if (global._setGlobalConsole === undefined) {
...
+
 export const checkPluginState = () => {
-  if (
-    !(() => {
-      'worklet';
-    }).__workletHash &&
-    !process.env.JEST_WORKER_ID
-  ) {
-    throw new Error(
+  if (!testWorklet.__workletHash && !process.env.JEST_WORKER_ID) {
```
Comment (satya164) [inline src/reanimated2/core.js:23]: Unrelated to this PR, but is it possible to expose this in a public API? My use case is to do feature detection regarding if I can use the Reanimated 2 APIs and then use the new implementation if possible.

File: src/reanimated2/core.js [hunk 2]
```diff
@@ -277,15 +268,23 @@ export function createAnimatedPropAdapter(adapter, nativeProps) {
...
-  const console = {
-    debug: runOnJS(capturableConsole.debug),
-    log: runOnJS(capturableConsole.log),
-    warn: runOnJS(capturableConsole.warn),
-    error: runOnJS(capturableConsole.error),
-    info: runOnJS(capturableConsole.info),
-  };
-  _setGlobalConsole(console);
-})();
+if (!NativeReanimated.useOnlyV1) {
```
Comment (satya164) [inline src/reanimated2/core.js:271]: Unrelated to this PR, but is it possible to expose this in a public API? My use case is to do feature detection regarding if I can use the Reanimated 2 APIs and then use the new implementation if possible.

Comment (piaskowyk) [reply src/reanimated2/core.js:271]: @Szymon20000 What do you think?

Comment (Szymon20000) [reply src/reanimated2/core.js:279]: Sure

**Pull request merged**